src/debug.c

```diff
@@ -1210,14 +1210,15 @@ static void* getAndSetMcontextEip(ucontext_t *uc, void *eip) {
     } \
     return old_val; \
 } while(0)
-#if defined(__APPLE__) && !defined(MAC_OS_10_6_DETECTED)
-    /* OSX < 10.6 */
+#if defined(__APPLE__) && defined(__POWERPC__)
+    /* OSX on PowerPC */
+    GET_SET_RETURN(uc->uc_mcontext->__ss.__srr0, eip);
+#elif defined(__APPLE__) && !defined(MAC_OS_10_6_DETECTED)
+    /* OSX < 10.6 on x86 */
     #if defined(__x86_64__)
     GET_SET_RETURN(uc->uc_mcontext->__ss.__rip, eip);
     #elif defined(__i386__)
     GET_SET_RETURN(uc->uc_mcontext->__ss.__eip, eip);
-    #else
-    GET_SET_RETURN(uc->uc_mcontext->__ss.__srr0, eip);
```

Comment:
forget to revert?

Reply:
This is PowerPC code, it moved above. There is no arm on < 10.5.

Reply:
Please help me to point out whether my understanding is wrong.
1. `MAC_OS_10_6_DETECTED` is always false in POWERPC?
   if so, do we only need to fix the detective of `MAC_OS_10_6_DETECTED` in POWERPC?
2. `uc->uc_mcontext->__ss.__srr0` is not only for POWERPC?

Reply:
1. No, it is not, because 10.6 supports ppc, so it is perfectly possible to build for ppc on 10.6. I have no idea why this specific code needs to check for 10.6 at all. It seems redundant. However the minimal change just to unbreak ppc is to check for it first (10.5 and 10.6 behave identically here).
I am not an expert on Intel, but this version leaves Intel code unchanged.

2. That ucontext line is valid only for ppc/ppc64 and should not be used elsewhere. srr0 is a PowerPC hardware register. It does not exist on x86 or arm.
A random example of Apple code using registers on ppc: https://opensource.apple.com/source/xnu/xnu-1504.7.4/osfmk/mach/ppc/_structs.h.auto.html